src/serac/physics/materials/thermal_expansion_material.hpp

```diff
@@ -0,0 +1,160 @@
+   */
+  virtual void modifyDisplacementGradient(mfem::DenseMatrix& du_dX) = 0;
+
+protected:
+  /**
+   * @brief Non-owning pointer to the reference element to stree-free configuration (target) transformation
+   *
+   */
+  mfem::ElementTransformation* parent_to_reference_transformation_;
+};
+
+/**
+ * @brief A simple isotropic finite deformation thermal expansion module.
+ *
+ * @details This implements the split deformation gradient model
+ *
+ * \f[
+ * \mathbf{F} = \mathbf{F}_\theta \mathbf{F}_M
+ * \f]
+ *
+ * where \f$\mathbf{F}\f$ is the total deformation gradient, \f$\mathbf{F}_\theta\f$ is the
+ * deformation gradient induced by thermal expansion, and \f$\mathbf{F}_M\f$ is the mechanical
+ * deformation gradient. The thermal deformation gradient of this model is given by
+ *
+ * \f[
+ * \mathbf{F}_\theta = \mathbf{I} + \alpha \left( \theta - \theta_\textrm{ref} \right)
+ * \f]
+ *
+ * where \f$\alpha\f$ is the coefficient of thermal expansion, \f$\theta\f$ is the current
+ * temperature, and \f$\theta_\textrm{ref}\f$ is the reference temperature.
+ */
+class IsotropicThermalExpansionMaterial : public ThermalExpansionMaterial {
+public:
+  /**
+   * @brief Construct a new Isotropic Thermal Expansion Material object
+   *
+   * @param coef_thermal_expansion The coefficient of thermal expansion \f$\alpha\f$
+   * @param reference_temp The reference temperature \f$\theta_{ref}\f$
+   * @param temp The current temperature \f$\theta\f$
+   */
+  IsotropicThermalExpansionMaterial(std::unique_ptr<mfem::Coefficient>&& coef_thermal_expansion,
+                                    std::unique_ptr<mfem::Coefficient>&& reference_temp, FiniteElementState& temp)
+      : c_coef_thermal_expansion_(std::move(coef_thermal_expansion)),
+        c_reference_temp_(std::move(reference_temp)),
+        temp_state_(temp)
+  {
+  }
+
+  /**
+   * @brief Modify the displacement gradient to include thermal strain
+   *
+   * @param[inout] du_dX The unmodified displacement gradient
+   * @pre The underlying element transformation must be set to the correct integration point
+   * for the coefficient evaluations to work correctly
+   */
+  void modifyDisplacementGradient(mfem::DenseMatrix& du_dX);
+
+  /**
+   * @brief Destroy the isotropic thermal expansion object
+   *
+   */
+  virtual ~IsotropicThermalExpansionMaterial() = default;
+
+  /**
+   * @brief Disable the default constructor
+   *
+   */
+  IsotropicThermalExpansionMaterial() = delete;
+
+protected:
+  /**
+   * @brief Coefficient of thermal expansion in constant form
+   */
+  mutable double coef_thermal_expansion_;
+
+  /**
+   * @brief Reference temperature in constant form
+   */
+  mutable double reference_temp_;
+
+  /**
+   * @brief Current temperature in constant form
+   */
+  mutable double temp_;
+
+  /**
+   * @brief Coefficient of thermal expansion in coefficient form
+   */
+  std::unique_ptr<mfem::Coefficient> c_coef_thermal_expansion_;
```

Comment:
do we need both `coef_thermal_expansion_` and `c_coef_thermal_expansion` to be members? It looks like maybe `EvalCoeffs()` could just return the thermal expansion coefficient at that point (`coef_thermal_expansion_`), rather than writing/reading to a member variable to access it.

Reply:
A fair point. I was following the pattern of the hyperelastic model.